SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/Storage.cs

```diff
@@ -349,12 +349,10 @@ private Task AddTaskToQueue(Action action)
                 {
                     action();
                 }
-                catch (Exception e) when (!(e is StorageException))
+                catch (Exception e)
                 {
-                    // Tasks should already be throwing only storage exceptions, but in case any are missed, 
-                    // which has happened, catch them here and wrap in a storage exception. This will prevent 
-                    // the exception from being unobserved.
-                    throw new StorageException(e);
+                    // If we use await on a task that returns an exception, it makes the app crash, however, using the awaiter works...
```

Comment:
Is it documented behavior?

Reply:
What I meant by that is, change this line by `throw await HandleStorageRelatedExceptionAsync(e)` and that will crash the app instead of rethrowing.

I don't know why. But anyway the throw await does not compile on the Func variant.

Reply:
![Capture](https://user-images.githubusercontent.com/623672/68909927-cc798100-0704-11ea-9f5b-a2972c82b30c.PNG)

For the func variant.

Reply:
In the action variant, it crashed instead of being caught:

```
"exception": {
    "type": "Microsoft.AppCenter.Storage.StorageException",
    "message": "The storage operation failed",
    "stackTrace": "   at Microsoft.AppCenter.Storage.Storage.<>c__DisplayClass24_0.<<AddTaskToQueue>b__0>d.MoveNext()\r\n   at Windows.ApplicationModel.Core.UnhandledError.Propagate()\r\n   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelper.<.ctor>b__17_1(Object sender, UnhandledErrorDetectedEventArgs eventArgs)",
    "innerExceptions": [
      {
        "type": "System.Exception",
        "message": "test",
        "stackTrace": "   at Microsoft.AppCenter.Storage.Storage.<>c__DisplayClass24_0.<<AddTaskToQueue>b__0>d.MoveNext()"
      }
    ]
  },
```

Reply:
Did you try `throw (await HandleStorageRelatedExceptionAsync(e))`? Sometimes compound statements with `await` require parentheses

Reply:
I tried, and it offers to remove parentheses, does not change.